basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -3416,4 +3422,18 @@ public static String getKey(Map<String, Object> tool, Map<String, Object> conten
 		}
 		return key;
 	}
+
+	public static void ensureGradebook(String context) {
+
+		GradebookFrameworkService gfs = (GradebookFrameworkService) ComponentManager.get("org.sakaiproject.service.gradebook.GradebookFrameworkService");
+		if (gfs != null) {
+			if (!gfs.isGradebookDefined(context)) {
+				try {
+					gfs.addGradebook(context, context);
+				} catch (GradebookExistsException e) { /* Can't happen */ }
```

Comment:
If this exception can't happen then a warning message saying that might be better?